unittests/src/main/java/uk/ac/manchester/tornado/unittests/batches/TestBatches.java

```diff
@@ -76,8 +77,19 @@ public static void compute(short[] arrayA, short[] arrayB, short[] arrayC) {
     @Test
     public void test100MB() {
 
+        long maxAllocMemory = getTornadoRuntime().getDefaultDevice().getDeviceContext().getMemoryManager().getHeapSize();
+
+        //check if there is enough memory for at least one chunk
+        if (maxAllocMemory < 100 * 1024 * 1024) {
+            throw new UnsupportedConfigurationException("Not enough memory to run the test");
+        }
+
```

Comment:
Create a method for this - You can pass the MBs required for each test. 

The method can be private and located in this class.

Reply:
We can also make the MBs and GBs constants.

Reply:
Yep, I put it this way, since it is a different setup for each test, but I will extract it